recipes/magics-metview/meta.yaml

```diff
@@ -0,0 +1,79 @@
+{% set name = "Magics" %}
+{% set version = "4.2.0" %}
+
+package:
+  name: magics-metview
+  version: {{ version }}
+
+source:
+  url: https://confluence.ecmwf.int/download/attachments/3473464/{{ name }}-{{ version }}-Source.tar.gz
+  sha256: d45701515d3329e1b3944d535c4d19f834d22a747a30da1d8c88b80affeb70cd
+  patches:
+    - 0001-Find-proj_6_0.lib-if-proj.lib-isn-t-found.patch
+    - 0002-Fix-for-Windows-build.patch
+
+build:
+  number: 0
+  skip: True  # [win]
+  detect_binary_files_with_prefix: true
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    # the CDTs below are needed when building with Metview enabled
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - {{ cdt('mesa-dri-drivers') }}  # [linux]
+    - {{ cdt('libselinux') }}  # [linux]
+    - {{ cdt('libxdamage') }}  # [linux]
+    - {{ cdt('libxfixes') }}  # [linux]
+    - {{ cdt('libxxf86vm') }}  # [linux]
+    - {{ cdt('libxcb') }}  # [linux]
+  host:
+    - pkg-config
+    - boost
+    - expat
+    - glib
+    - pango
+    - cairo
+    - proj
+    - eccodes
+    - libnetcdf
+    - jinja2
+    - qt
+    - zlib
+  run:
+    - python
+    - expat
+    - pango
+    - cairo
+    - proj
+    - eccodes
+    - simplejson
+    - libnetcdf
+    - qt
+    - zlib
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+  run_constrained:
+    # magics-metview is incompatible with magics
+    - magics ==9999999999
+
+test:
+  commands:
+    - printf "%s\n" "${MAGPLUS_HOME:?MAGPLUS_HOME is not set}"
```

Comment:
Can you do anything more for testing that the build works like (in order of preference):
* running commands
* running `command --help` for each binary
* Using `which command` to make sure it's on `$PATH`

If you see failures due to the lack of X11 server on Linux you can workaround it using the suggestion in https://github.com/conda-forge/staged-recipes/pull/8645#issuecomment-505879846

Reply:
Thanks for the thorough review, @chrisburr ! Sadly I don't think there are any testable commands which can be run here. The main output of this build is a trio of libraries (libMagPlusDouble.so, libMagPlusSingle.so, libMagPlus.so on Linux). A couple of binaries are produced but I think they're legacy and not really used. If it helps, there is some testing with `ctest` during the build script.

Reply:
I was wrong ;) Added a test for one of the binaries.

Reply:
As it mainly provides shared libraries, should this recipe use `run_exports` to ensure that if it is used in a downstream package's `host` requirements, it is automatically added to `run` with appropriate version constrains? 

For example:
```yaml
build:
  run_exports:
    - {{ pin_subpackage('magics-metview', max_pin='x.x') }}
```
(Adjust `x.x` to `x` or `x.x.x` depending on the expected ABI stability)

See https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements

Reply:
You can also use `test -f $PREFIX/lib/libSomething.so` to make sure libraries are installed in the right place as expected.

(I've accidentally deployed empty packages when skipping these tests so they are useful.)

Reply:
Never heard of `run_exports` before - looks cool. I've updated and added a few `test -f`s. Thanks :)